compiler/lib/Dialect/mhlo/Transforms/CanonicalizeExt.cpp

```diff
@@ -1377,6 +1377,47 @@ struct FoldLargeConcatenate : public OpRewritePattern<mhlo::ConcatenateOp> {
   }
 };
 
+struct CanonicalizeClamp : public OpRewritePattern<mhlo::ClampOp> {
+  using OpRewritePattern<mhlo::ClampOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(mhlo::ClampOp op,
+                                PatternRewriter &rewriter) const override {
+    auto minConst = op.getMin().getDefiningOp<mhlo::ConstantOp>();
+    auto maxConst = op.getMax().getDefiningOp<mhlo::ConstantOp>();
+    if (!minConst || !maxConst) {
+      return failure();
+    }
+    auto minAttr =
+        dyn_cast_if_present<DenseElementsAttr>(minConst.getValueAttr());
+    auto maxAttr =
+        dyn_cast_if_present<DenseElementsAttr>(maxConst.getValueAttr());
+    if (!minAttr || !maxAttr) {
+      return failure();
+    }
+    if (!minAttr.isSplat() || !maxAttr.isSplat()) {
+      return failure();
+    }
+    // replace splat const with scalar
+    if (minAttr.getType().getRank() > 0 && maxAttr.getType().getRank() > 0) {
+      minAttr = minAttr.dyn_cast<SplatElementsAttr>().resizeSplat(
```

Comment:
nit: => `dyn_cast<SplatElementsAttr>(minAttr).resizeSplat`.
Since llvm has marked the method `.cast` `.dyn_cast` as deprecated.